flaml/model.py

```diff
@@ -574,14 +570,25 @@ def tokenizer(self):
     def data_collator(self):
         from .nlp.huggingface.data_collator import task_to_datacollator_class
 
-        return (
-            task_to_datacollator_class[self._task](
-                tokenizer=self.tokenizer,
-                pad_to_multiple_of=8,  # if self._training_args.fp16 else None,
-            )
-            if self._task in (MULTICHOICECLASSIFICATION, TOKENCLASSIFICATION)
-            else None
-        )
+        data_collator_class = task_to_datacollator_class.get(self._task)
+
+        if data_collator_class:
+            kwargs = {
+                "model": self._model_init(),  # need to set model, or there's ValueError: Expected input batch_size (..) to match target batch_size (..)
+                "label_pad_token_id": -100,
+                "pad_to_multiple_of": 8,

```

Comment:
If there is a non-trivial reason of using these values, write comments. I can't tell whether they are right.